prow/config/config.go

```diff
@@ -96,6 +96,7 @@ type ProwConfig struct {
 	Gerrit           Gerrit           `json:"gerrit,omitempty"`
 	GitHubReporter   GitHubReporter   `json:"github_reporter,omitempty"`
 	SlackReporter    *SlackReporter   `json:"slack_reporter,omitempty"`
+	InRepoConfig     InRepoConfig     `json:"in_repo_config"`
```

Comment:
Sidenote: What about starting to name the whole thing `DynamicConfig` instead?

Reply:
Your call!